streams/src/main/java/org/apache/kafka/streams/state/internals/CompositeReadOnlyWindowStore.java

```diff
@@ -80,4 +80,32 @@ public CompositeReadOnlyWindowStore(final StateStoreProvider provider,
                                                                provider.stores(storeName, windowStoreType).iterator(),
                                                                nextIteratorFunction));
     }
+    
+    @Override
+    public KeyValueIterator<Windowed<K>, V> all() {
+        final NextIteratorFunction<Windowed<K>, V, ReadOnlyWindowStore<K, V>> nextIteratorFunction = new NextIteratorFunction<Windowed<K>, V, ReadOnlyWindowStore<K, V>>() {
+            @Override
+            public KeyValueIterator<Windowed<K>, V> apply(final ReadOnlyWindowStore<K, V> store) {
+                return store.all();
+            }
+        };
+        return new DelegatingPeekingKeyValueIterator<>(storeName,
+                new CompositeKeyValueIterator<>(
+                        provider.stores(storeName, windowStoreType).iterator(),
+                        nextIteratorFunction));
+    }
+    
+    @Override
+    public KeyValueIterator<Windowed<K>, V> fetchAll(final long timeFrom, final long timeTo) {
```

Comment:
I realize this wasn't in existing methods, but do we want to add a check that `timeFrom` <= `timeTo` ?

Reply:
The question is where should we do this check: I think we should just do the check on the outer-most classes. Maybe we can introduce another abstract class similar to `AbstractStateStore` which does the checking. But I think it is rather bigger range than this PR and can be done in a separate JIRA.